### PR TITLE
More gun variants, part one

### PR DESCRIPTION
**Pull request by DoctorBoomstick**
#### Summary
None

#### Purpose of change
Ever since the gun variants system has been implemented, there has remained a large discrepancy in the number of guns present within CDDA’s portfolio that make use of this feature, and those that do not. As more and more fun murder sticks are included, this issue becomes a little bit more pronounced with every new and obscure weapon that gets added. Whilst this is the fault of no person, save for possibly those silly little people out and about in the world making new fun murder sticks which must sooner or later be simulated in the game, the fact that many weapons, save for all the shotguns and a modest collection of rifles, pistols, and machineguns, still don’t use the variants system, I personally feel, is lessening the overall point and impact that this feature provides. I honestly don’t know how many people make use of this feature, or how useful it is considering that we have the generic weapons mod, but I am of the mindset that the feature is here, may as well make use of it.

#### Describe the solution
For this PR, I’ve started off on a light footing. I’ve added variants to the .270 Winchester Remington 700, the S&W 500, the Big Horn Model 89, the Mosin M44, and the Mosin 91/30. On this latter note, I’ve melded the EBR iterations of both the latter guns into variants, with contained grip, rail, and stock mount gun mods providing their different modular capabilities. On a completely unrelated note, I also went ahead and included an integral folding bayonet to the M44 Mosin, for the two important reasons that the real guns come equipped with one and having big stabbing sticks on the end of Russian guns sounds like fun. When all things are considered, this completes adding variants for all the .270, .500, and 7.62x54mm guns currently in the game.

#### Describe alternatives you've considered
I considered first completing all the variant work for all calibres before putting forwards this pr suggestion. The reasons that I decided against this is simply because it will take a preposterous amount of time to add variants for potentially all the guns and having a steady trickle of completed calibres sounded more in line with preserving my sanity. In addition, I believe working with small, manageable bite sized pieces just makes everything simpler for all parties involved.
Apart for this, I was flirting with the idea of making the Mosin M44 and 91/30 EBR rifles have their own respective variants, rather than being lumped in with the plane non-modern iterations of the guns. I decided not to do this as the method I used cuts down on the number of similar guns present for the player, thus making stuff simpler. Sure, I could have made the “refurbished Mosin carbine,” and the “refurbished Mosin battle rifle,” to go with the plane Mosin carbine and battle rifle, but do we really need four similarly named guns which pretty much do the same things? Plus, the differences between them, the former’s capability for mounting accessories, can and is reinstated by adding contained mods. The only sign, from a gameplay perspective, that something is different is that, by the nature of how the variant system works, the stats for both the Mosin M44 and M44 EBR, and the 91/30 and 91/30 EBR are the same. Outside of their accessary capabilities, the normal and EBR iterations are essentially identical. That being said, it would be nice if down the line, variant guns of this nature could also define their own stats.
#### Testing
All elements of the variant change have been tested, I.E., spawning in the effected firearms and checking them out.

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
762R guns before this change:
| id | weight | volume | length | ammotype | modes | magazines | dispersion | recoil | damage | 
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | 
| obrez | 3000 | 2220 | 400 | 762R | 1 | 5 | 90 | 0 | 0 | 
| mosin44 | 4100 | 2720 | 1015 | 762R | 1 | 5 | 90 | 0 | 0 | 
| mosin44_ebr | 4330 | 2870 | 1015 | 762R | 1 | 5 | 90 | 0 | 0 | 
| mosin91_30 | 4000 | 3120 | 1241 | 762R | 1 | 5 | 90 | 0 | 0 | 
| mosin91_30_ebr | 4230 | 3300 | 1241 | 762R | 1 | 5 | 90 | 0 | 0 |

File: data/json/items/gun/500.json [hunk 1]
```diff
@@ -43,8 +52,17 @@
     "copy-from": "pistol_revolver",
     "looks_like": "sw_619",
     "type": "GUN",
-    "name": { "str_sp": "S&W 500" },
-    "description": "The 5-shot Smith and Wesson 500 revolver fires the comparably-named .500 S&W Magnum.  It's an impressive weapon.",
+    "name": { "str": ".500 magnum revolver" },
```
Comment (TheShadowFerret) [inline data/json/items/gun/500.json:55]:
```suggestion
    "name": { "str": ".500 Magnum revolver" },
```

File: data/json/items/gun/500.json [hunk 2]
```diff
@@ -4,8 +4,17 @@
     "copy-from": "rifle_manual",
     "looks_like": "ar15",
     "type": "GUN",
-    "name": { "str_sp": "Big Horn Model 89" },
-    "description": "Built in the image of the venerable Winchester Model 1886, Big Horn Armory's Model 89 rifle packs all the power of the .500 S&W Magnum cartridge into a package less likely to break one's wrist.",
+    "name": { "str": ".500 lever gun" },
```
Comment (TheShadowFerret) [inline data/json/items/gun/500.json:7]:
```suggestion
    "name": { "str": ".500 Magnum lever gun" },
```

File: data/json/items/gun/270win.json [hunk 3]
```diff
@@ -4,8 +4,17 @@
     "copy-from": "rifle_manual",
     "looks_like": "ar15",
     "type": "GUN",
-    "name": { "str_sp": "Remington 700 .270 Win" },
-    "description": "A classic bolt-action rifle chambered for .270 Winchester, very popular among hunters.  This is a CDL SF model with a forged, fluted, 416 stainless steel barrel threaded into the receiver and a recessed bolt face.  It has checkered walnut furniture and a recoil pad to reduce perceived recoil.",
+    "name": { "str": "bolt-action hunting rifle" },
+    "description": "Featuring a hansom set of wooden furnishings, a classic design, and a no nonsense .270 calibre chambering, this is a powerful hunting rifle.  With a tried and proven pedigree dating back over 60 years, millions of such rugged arms have been made, exported, and used throughout the United States and the world at large.  Where hunting is concerned, this is a highly capable and well proven weapon in all respects.  Fortunate, given that your prey in these dark days may turn out to be a little bit more feisty than typical White-Tailed deer, moose, and black bears.",
```
Comment (anothersimulacrum) [inline data/json/items/gun/270win.json:8]:
```suggestion
    "description": "Featuring a handsome set of wooden furnishings, a classic design, and a no nonsense .270 caliber chambering, this is a powerful hunting rifle.  With a tried and proven pedigree dating back over 60 years, millions of such rugged arms have been made, exported, and used throughout the United States and the world at large.  Where hunting is concerned, this is a highly capable and well proven weapon in all respects.  Fortunate, given that your prey in these dark days may turn out to be a little bit more feisty than typical White-Tailed deer, moose, and black bears.",
```

**Pull request merged**